backend/pkg/api/handle_frontend.go

```diff
@@ -1,28 +1,43 @@
 package api
 
 import (
+	"bytes"
+	"fmt"
 	"io/ioutil"
 	"net/http"
 	"os"
 
 	"go.uber.org/zap"
 )
 
-// getIndexFile loads and prepares the template index.html.
-// Before returning, this method injects the needed server variables (like
-// for example __COMMIT_SHA__) so the file is ready to be sent to a connecting browser.
-func (api *API) getIndexFile(filePath string) ([]byte, error) {
-	indexPath := filePath + "/index.html"
-	index, err := ioutil.ReadFile(indexPath)
-	if err != nil {
-		return nil, err
+var (
+	srcAttribute  = []byte(`src="/`)
+	hrefAttribute = []byte(`href="/`)
+	baseURLMarker = []byte(`__BASE_URL_REPLACE_MARKER__`)
+)
+
+// handleGetIndex returns the SPA (index.html)
+func (api *API) handleGetIndex(index []byte) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// check for 'X-Forwarded-Prefix' and if set,
+		// add it as a prefix to all occurences of    (src="/)|(href="/)
+		if xForwardedPrefixAr, exists := r.Header["X-Forwarded-Prefix"]; exists && len(xForwardedPrefixAr) > 0 {
+			xForwardedPrefix := xForwardedPrefixAr[0]
```

Comment:
```suggestion
		forwardedPrefix := r.Header.Get("X-Forwarded-Prefix")
		if forwardedPrefix != "" {
			xForwardedPrefix := forwardedPrefix[0]
```

Reply:
I initially used `r.Header.Get` but the description of that function says

> Get gets the first value associated with the given key. If
there are no values associated with the key, Get returns "".
It is case insensitive; **textproto.CanonicalMIMEHeaderKey is
used to canonicalize the provided key. To use non-canonical keys,
access the map directly.**

And I wasn't able to quickly find a list of which headers are canonical and which aren't.

Reply:
applied